CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+
+## <a name="issue"></a> Found a Bug?
+If you find a bug in the source code, you can help us by
+[submitting an issue](https://github.com/UdacityMobileWebScholarship/showcase-app/issues/new) to our GitHub Repository. Even better, you can check out issues [open for claim](https://github.com/UdacityMobileWebScholarship/showcase-app/labels/open%20for%20claim) and claim one!
+
+## <a name="feature"></a> Missing a Feature?
+You can *request* a new feature by [submitting an issue](https://github.com/UdacityMobileWebScholarship/showcase-app/issues/new) to our GitHub
+Repository. Given the project timeline and necessity for the feature, we'll discuss the roadmap accordingly. If you would like to *implement* a new feature, please submit an issue with
+a proposal for your work first, to be sure that we can use it.
+
+## <a name="submit"></a> Submission Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
+Before you submit an issue, please search the issue tracker, maybe an issue for your identified problem already exists and you can join the discussion there only.
+For bugs, please try to provide as much information possible to reproduce it as possible, that will really help us a lot!
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+1. Make sure there is an issue open for the PR and it is assigned to you before starting any work. That way it'll be helpful to eliminate duplicate efforts.
+2. Fork the repo.
+3. Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch master
+     ```
+4. Create your patch, **including appropriate test cases**.
+5. Run the full project test suite, as described in the [readme](https://github.com/UdacityMobileWebScholarship/showcase-app/tree/master#user-content-project-set-up),
+  and ensure that all tests pass.
+6. Commit your changes using a descriptive commit message. ex-
+```shell
+    git commit -m "doc: update readme"
+    git commit -m "ref: refactor API service"
+    git commit -m "feat: add authentication service"
+```
+note: it is not mandatory to write commit messages in the same format, but if you'd like to add labels in the message, that'll really help with understanding the purpose of the commit.
+Might be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: webpack, npm, deploy script)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Travis, Circle)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix
+* **perf**: A code change that improves performance
+* **ref**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* **test**: Adding missing tests or correcting existing tests
+* **UI**: UI related additions
```

Comment:
I think we should not add labels to the commit messages as the history sometimes looks weird while looking at very early commits. This categorization should ideally be done at the PR level.

Reply:
This is optional as mentioned.